WordPress/UITestsFoundation/Screens/Editor/AztecEditorScreen.swift

```diff
@@ -11,63 +13,44 @@ public class AztecEditorScreen: BaseScreen {
     }
 
     let mode: Mode
-    var textView: XCUIElement
-
-    private var richTextField = "aztec-rich-text-view"
-    private var htmlTextField = "aztec-html-text-view"
-
-    let editorCloseButton = XCUIApplication().navigationBars["Azctec Editor Navigation Bar"].buttons["Close"]
-    let publishButton = XCUIApplication().buttons["Publish"]
-    let publishNowButton = XCUIApplication().buttons["Publish Now"]
-    let moreButton = XCUIApplication().buttons["more_post_options"]
-    let uploadProgressBar = XCUIApplication().progressIndicators["Progress"]
-
-    let titleView = XCUIApplication().textViews["Title"]
-    let contentPlaceholder = XCUIApplication().staticTexts["aztec-content-placeholder"]
-
-    let mediaButton = XCUIApplication().buttons["format_toolbar_insert_media"]
-    let insertMediaButton = XCUIApplication().buttons["insert_media_button"]
-    let headerButton = XCUIApplication().buttons["format_toolbar_select_paragraph_style"]
-    let boldButton = XCUIApplication().buttons["format_toolbar_toggle_bold"]
-    let italicButton = XCUIApplication().buttons["format_toolbar_toggle_italic"]
-    let underlineButton = XCUIApplication().buttons["format_toolbar_toggle_underline"]
-    let strikethroughButton = XCUIApplication().buttons["format_toolbar_toggle_strikethrough"]
-    let blockquoteButton = XCUIApplication().buttons["format_toolbar_toggle_blockquote"]
-    let listButton = XCUIApplication().buttons["format_toolbar_toggle_list_unordered"]
-    let linkButton = XCUIApplication().buttons["format_toolbar_insert_link"]
-    let horizontalrulerButton = XCUIApplication().buttons["format_toolbar_insert_horizontal_ruler"]
-    let sourcecodeButton = XCUIApplication().buttons["format_toolbar_toggle_html_view"]
-    let moreToolbarButton = XCUIApplication().buttons["format_toolbar_insert_more"]
-
-    let unorderedListOption = XCUIApplication().buttons["Unordered List"]
-    let orderedListOption = XCUIApplication().buttons["Ordered List"]
-
-    // Action sheets
-    let actionSheet = XCUIApplication().sheets.element(boundBy: 0)
-    let postSettingsButton = XCUIApplication().sheets.buttons["Post Settings"]
-    let keepEditingButton = XCUIApplication().sheets.buttons["Keep Editing"]
-    let postHasChangesSheet = XCUIApplication().sheets["post-has-changes-alert"]
-
-    init(mode: Mode) {
-        var textField = ""
+    private(set) var textView: XCUIElement
+
+    private let richTextField = "aztec-rich-text-view"
+    private let htmlTextField = "aztec-html-text-view"
+
+    var mediaButton: XCUIElement { app.buttons["format_toolbar_insert_media"] }
+    var sourcecodeButton: XCUIElement { app.buttons["format_toolbar_toggle_html_view"] }
+
+    private let textViewGetter: (String) -> (XCUIApplication) -> XCUIElement = { identifier in
+        return { app in
```

Comment:
We don't see this syntax much: It's a closure that returns another closure.

You can parse it like this

```
let textViewGetter: (Input) -> (Output)

let textViewGetter: (String) -> (Output)

let textViewGetter: (String) -> (XCUIApplication) -> XCUIElement
```

That's why we have

```swift
{ identifier in
  return app in 
```

`{ identifier in return ... }` is the `(String) -> (Output)`, `{ app in ... }` is the `(Output)` value, i.e. `XCUIApplication -> XCUIElement`.

---

I wrote this to generate the getter closure based on a `textField` value that becomes known only at runtime, as you can see in the `init` code just below.